core/service/section/container-section-service.js

```diff
@@ -71,6 +73,26 @@ exports.ContainerSectionService = AbstractSectionService.specialize({
         value: function (container) {
             return this._containerRepository.saveContainer(container);
         }
+    },
+
+    pullImageToContainer: {
+        value: function (imageName, containerId) {
+            var self = this;
+
+            this._middlewareTaskRepository.getNewMiddlewareTaskWithNameAndArgs("docker.image.pull", [imageName, containerId]).then(function(middlewareTask) {
+                return self._middlewareTaskRepository.runMiddlewareTask(middlewareTask);
+            });
+        }
+    },
+
+    deleteImageFromContainer: {
+        value: function (imageName, containerId) {
```

Comment:
This should be called `pullImageToDockerHost` and should take `dockerHostId` as an argument.